lib/_tokenUtils.ts

```diff
@@ -68,6 +68,17 @@ export function setUserToken(
   return this._userToken;
 }
 
+export function setAuthenticatedUserToken(
+  this: AlgoliaAnalytics,
+  authenticatedUserToken: number | string
+): number | string {
+  this._authenticatedUserToken = authenticatedUserToken;
+  if (isFunction(this._onAuthenticatedUserTokenChangeCallback)) {
+    this._onAuthenticatedUserTokenChangeCallback(this._authenticatedUserToken);
+  }
+  return this._authenticatedUserToken;
+}
+
```

Comment:
I think that we should add a getter too, as we already have it for `userToken`

Reply:
My thinking for leaving it out was that the `authenticatedUserToken` is always provided, unlike the `userToken` that may be set by us, so I wasn't sure that a getter was necessary. Though, I can see how it might be useful if they were to lose the token on their end somehow. Or maybe even just to check that it's been set properly?

Reply:
I don't know exactly how or why it would be used, but I'd bet we'd have a feature request for it in fairly short order if it wasn't available.